src/factories/routePathFactory.ts

```diff
@@ -2,17 +2,24 @@ import { INode, IRoutePath } from '../models';
 import NodeFactory from './nodeFactory';
 
 class RoutePathFactory {
+
     // suunta to IRoutePath
     public static createRoutePath = (suunta: any, isVisible:boolean): IRoutePath => {
         const nodes:INode[]
         = suunta.reitinlinkkisByReitunnusAndSuuvoimastAndSuusuunta.edges
-            .map((node: any) => NodeFactory.createNode(node.node));
+            .map((node: any) => {
+                return NodeFactory.createNode(node.node);
+            });
+
+        const geoJson = JSON.parse(suunta.geojson);
+        const asd = NodeFactory.createStartingPointNode(geoJson.coordinates[0]);
```

Comment:
nodes.push(NodeFactory.createStartingPointNode(geoJson.coordinates[0]);?

The order of nodes shouldn't matter when they have NodeType that separates them. Just prefer using push over unshift